test/tools/test_tools_memories.py

```diff
@@ -58,3 +62,415 @@ def test_sopel_identifier_memory_channel_str():
     assert memory[channel] == test_value
     assert memory['#adminchannel'] == test_value
     assert memory['#AdminChannel'] == test_value
+
+
+def test_sopel_identifier_memory_setdefault():
+    memory = memories.SopelIdentifierMemory()
+    assert 'foo' not in memory
+
+    memory.setdefault('Foo')
+    assert len(memory) == 1
+    assert 'foo' in memory
+    assert 'fOO' in memory
+    assert memory['fOo'] is None
+
+    memory.setdefault('bAr', 'DEFAULT')
+    assert len(memory) == 2
+    assert 'FoO' in memory
+    assert 'BaR' in memory
+    assert memory['Bar'] == 'DEFAULT'
+
+    assert memory.setdefault('baR', 'different') == 'DEFAULT'
+    assert len(memory) == 2
+
+
+def test_sopel_identifier_memory_copy():
+    memory = memories.SopelIdentifierMemory()
+    memory['SomeCamelCase'] = True
+    memory['loweronly'] = False
+    assert len(memory) == 2
+    assert isinstance(memory, memories.SopelIdentifierMemory)
+
+    copied = memory.copy()
+    assert len(copied) == 2
+    assert 'SomeCamelCase' in copied
+    assert 'loweronly' in copied
+    assert isinstance(memory, memories.SopelIdentifierMemory)
```

Comment:
This might need another step:

```python
    copied['newKey'] = 'value'
    assert 'newKey' in copied
    assert 'newKey' not in memory
```

Just to make sure that the "copy" actually create a copy, and not a reference to the original object.